src/net/mormot.net.ldap.pas

```diff
@@ -928,6 +953,43 @@ function AsnNext(var Pos: integer; const Buffer: TAsnObject;
     end;
 end;
 
+function DNToCannonical(DN: RawUtf8): RawUtf8;
+const
+  DomainComponent = 'DC';
+  OrganizationalUnit = 'OU';
+  CommonName = 'CN';
+var
+  P: PUtf8Char;
+  Domain, OU, Common, PartType, Value: RawUtf8;
+begin
+  if DN = '' then
+    Exit('');

```

Comment:
Exit('') is not compatible with old Delphi
- to be done in two steps: `result := ''; if DN = '' then exit;`